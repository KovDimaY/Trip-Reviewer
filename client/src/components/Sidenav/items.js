import * as routes from './../../constants/routes';

export const items = [
    {
        type: 'navItem',
        icon: 'home',
        text: 'Home',
        link: routes.HOME,
        restricted: false
    },
    {
        type: 'navItem',
        icon: 'user',
        text: 'My Profile',
        link: routes.USER_PROFILE,
        restricted: true
    },
    {
        type: 'navItem',
        icon: 'user-plus',
        text: 'Register',
        link: routes.REGISTER,
        restricted: false,
        exclude: true
    },
    {
        type: 'navItem',
        icon: 'sign-in-alt',
        text: 'Login',
        link: routes.LOGIN,
        restricted: false,
        exclude: true
    },
    {
        type: 'navItem',
        icon: 'file-alt',
        text: 'My reviews',
        link: routes.USER_REVIEWS,
        restricted: true
    },
    {
        type: 'navItem',
        icon: 'edit',
        text: 'Add reviews',
        link: routes.ADD_TRIP,
        restricted: true
    },
    {
        type: 'navItem',
        icon: 'sign-out-alt',
        text: 'Logout',
        link: routes.LOGOUT,
        restricted: true
    }
];