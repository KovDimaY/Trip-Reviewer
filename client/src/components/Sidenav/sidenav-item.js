import React from 'react';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';

const SidenavItems = ({ item }) => {
    return (
        <div className={item.type}>
            <Link to={item.link}>
                <FontAwesome name={item.icon}/>
                {item.text}
            </Link>
        </div>
    );
};

export default SidenavItems;