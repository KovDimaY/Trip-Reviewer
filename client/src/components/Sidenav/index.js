import React from 'react';
import ReactSideNav from 'react-simple-sidenav';

import { items } from './items';
import SidenavItem from './../../containers/SidenavItem/index';

const Sidenav = (props) => {
    return (
       <ReactSideNav
            showNav={props.showNav}
            onHideNav={props.onHideNav}
            navStyle={{
                background:'#242424',
                maxWidth:'220px'
            }}
       >    
            {
               items.map((item, i) =>  <SidenavItem item={item} key={i}/>) 
            }
        </ReactSideNav>
    );
};

export default Sidenav;