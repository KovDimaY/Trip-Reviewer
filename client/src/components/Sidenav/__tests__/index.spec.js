import React from 'react';
import { create } from 'react-test-renderer';
 
import Sidenav from './../../Sidenav';

jest.mock('react-simple-sidenav', () => 'ReactSideNav');
jest.mock('./../sidenav-item', () => 'SidenavItem');

const mockComponent = props => {
    return (
        <Sidenav {...props} />
    );
};
 
describe('<Sidenav />', () => {
    it('should render component', () => {
        const tree = create(mockComponent()).toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});