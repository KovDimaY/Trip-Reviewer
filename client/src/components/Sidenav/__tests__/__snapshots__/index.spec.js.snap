// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidenav /> should render component 1`] = `
<ReactSideNav
  navStyle={
    Object {
      "background": "#242424",
      "maxWidth": "220px",
    }
  }
  onHideNav={undefined}
  showNav={undefined}
>
  <SidenavItem
    item={
      Object {
        "icon": "home",
        "link": "/",
        "restricted": false,
        "text": "Home",
        "type": "navItem",
      }
    }
  />
  <SidenavItem
    item={
      Object {
        "icon": "user",
        "link": "/user",
        "restricted": true,
        "text": "My Profile",
        "type": "navItem",
      }
    }
  />
  <SidenavItem
    item={
      Object {
        "exclude": true,
        "icon": "user-plus",
        "link": "/user/register",
        "restricted": false,
        "text": "Register",
        "type": "navItem",
      }
    }
  />
  <SidenavItem
    item={
      Object {
        "exclude": true,
        "icon": "sign-in-alt",
        "link": "/login",
        "restricted": false,
        "text": "Login",
        "type": "navItem",
      }
    }
  />
  <SidenavItem
    item={
      Object {
        "icon": "file-alt",
        "link": "/user/user-reviews",
        "restricted": true,
        "text": "My reviews",
        "type": "navItem",
      }
    }
  />
  <SidenavItem
    item={
      Object {
        "icon": "edit",
        "link": "/trip/add",
        "restricted": true,
        "text": "Add reviews",
        "type": "navItem",
      }
    }
  />
  <SidenavItem
    item={
      Object {
        "icon": "sign-out-alt",
        "link": "/user/logout",
        "restricted": true,
        "text": "Logout",
        "type": "navItem",
      }
    }
  />
</ReactSideNav>
`;
